crates/bin/pd/src/migrate/testnet77.rs

```diff
@@ -0,0 +1,78 @@
+//! Contains functions related to the migration script of Testnet77.
+//! The Testnet 77 release included several consensus-breaking changes,
+//! but no state-breaking changes, so the migration is essentially a no-op,
+//! other than resetting the halt bit.
+use anyhow;
+use cnidarium::Storage;
+use jmt::RootHash;
+use penumbra_app::app::StateReadExt as _;
+use penumbra_sct::component::clock::EpochRead;
+use std::path::PathBuf;
+use tracing::instrument;
+
+use crate::testnet::generate::TestnetConfig;
+
+/// Run the full migration, given an export path and a start time for genesis.
+///
+/// Menu:
+/// - Reconstruct a correct VCB balance for the auction component.
+#[instrument]
+pub async fn migrate(
+    storage: Storage,
+    pd_home: PathBuf,
+    genesis_start: Option<tendermint::time::Time>,
+) -> anyhow::Result<()> {
+    // Setup:
+    let initial_state = storage.latest_snapshot();
+    let chain_id = initial_state.get_chain_id().await?;
+    let root_hash = initial_state
+        .root_hash()
+        .await
+        .expect("chain state has a root hash");
+    let pre_upgrade_root_hash: RootHash = root_hash.into();
+    let pre_upgrade_height = initial_state
+        .get_block_height()
+        .await
+        .expect("chain state has a block height");
+    let post_upgrade_height = pre_upgrade_height.wrapping_add(1);
```

Comment:
I think we're missing creating the `StateDelta` and calling `delta.ready_to_start()` + commit in place, unless my eyes betray me